core/src/main/java/discord4j/core/shard/GatewayBootstrap.java

```diff
@@ -798,6 +797,8 @@ public StoreService adaptStoreService(StoreService storeService) {
                     // wire internal shard coordinator events
                     // TODO: transition into separate lifecycleSink for these events
                     forCleanup.add(gatewayClient.dispatch()
+                            .filter(dispatch -> !dispatch.isFromGateway())
+                            .map(LazyDispatch::getData)
```

Comment:
This also throws some IDE warnings. Even if it's not a concern we should assert it being not null at some point.